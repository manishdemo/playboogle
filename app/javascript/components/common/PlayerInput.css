.word-submit {
    background-color: #95bb76;
    border-color: #97935c;
    margin-left: 10px;
    border-radius: 15px;
    height: 25px;
    width: 80px;
}

.word-submit:focus {
    outline: none;
    box-shadow: none;
}

.word-input {
    background-color: #8be3e1;
    border-color: #8be3e1;
    margin-left: 10px;
}
