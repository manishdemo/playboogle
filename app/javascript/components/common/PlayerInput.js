import React from "react";
import "./PlayerInput.css"

class PlayerInput extends React.Component {
    constructor(props) {
        super(props);

        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleChange(event) {
        this.props.onInputWordChange(event.target.value)
    }

    handleSubmit(event) {
        if (this.props.isGameStarted) {
            this.props.onWordSubmit(event);
        }
        event.preventDefault();
    }

    render() {
        const inputWord = this.props.inputWord;
        return (
            <div >
                <br/>
                <form onSubmit={this.handleSubmit}>
                    <label>
                        Your word:
                        <input type="text" value={inputWord} onChange={this.handleChange}
                               disabled = {(this.props.isGameStarted)? "" : "disabled"}
                               autofocus="true"
                               className="word-input" />
                    </label>
                    <input type="submit" value="Submit" className="word-submit" />
                </form>
            </div>
        );
    }
}

export default PlayerInput