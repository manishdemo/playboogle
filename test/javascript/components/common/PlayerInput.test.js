import React from "react";

import PlayerInput from "../../../../app/javascript/components/common/PlayerInput";
import {describe, expect, it} from "@jest/globals";
import {mount, shallow} from "enzyme";

describe('PlayerInput', () => {
    it('should render correctly', () => {
        const component = shallow(<PlayerInput inputWord="word" /> );
        expect(component).toMatchSnapshot();
    });

    it('handleWordSubmit should be called when the word submit button is pressed, while PlayerInput is in ' +
        'active mode', () => {
        const handleWordSubmitFn = jest.fn();
        const component = mount(<PlayerInput onWordSubmit={handleWordSubmitFn} isGameStarted={true}/>);
        component.find('.word-submit').first().simulate('submit');
        expect(handleWordSubmitFn).toHaveBeenCalled();
        component.unmount();

    });

    it('handleWordSubmit should not be called when the word submit button is pressed, while PlayerInput is ' +
        'in deactivated mode', () => {
        const handleWordSubmitFn = jest.fn();
        const component = mount(<PlayerInput onWordSubmit={handleWordSubmitFn} isGameStarted={false}/>);
        component.find('.word-submit').first().simulate('submit');
        expect(handleWordSubmitFn).toHaveBeenCalledTimes(0);
        component.unmount();
    });

});