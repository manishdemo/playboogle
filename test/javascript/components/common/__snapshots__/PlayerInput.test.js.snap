// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerInput should render correctly 1`] = `
<PlayerInput
  innerRef={null}
  inputWord="word"
/>
`;
